Rocket.Chat/Models/Subscription/UnmanagedSubscription.swift

```diff
@@ -0,0 +1,105 @@
+//
+//  UnmanagedSubscription.swift
+//  Rocket.Chat
+//
+//  Created by Samar Sunkaria on 8/10/18.
+//  Copyright © 2018 Rocket.Chat. All rights reserved.
+//
+
+import Foundation
+import DifferenceKit
+
+struct UnmanagedSubscription: UnmanagedObject, Equatable {
+    typealias Object = Subscription
+
+    var managedObject: Subscription
+    var identifier: String?
+    var privateType: String
+    var type: SubscriptionType
+    var rid: String
+    var name: String
+    var fname: String
+    var unread: Int
+    var userMentions: Int
+    var groupMentions: Int
+    var open: Bool
+    var alert: Bool
+    var favorite: Bool
+    var createdAt: Date?
+    var lastSeen: Date?
+    var roomTopic: String?
+    var roomDescription: String?
+    var roomReadOnly: Bool
+    var roomUpdatedAt: Date?
+    var roomLastMessage: Message?
+    var roomLastMessageText: String?
+    var roomLastMessageDate: Date?
+    var roomBroadcast: Bool
+    var roomOwnerId: String?
+    var otherUserId: String?
+    var disableNotifications: Bool
+    var hideUnreadStatus: Bool
+    var desktopNotificationDuration: Int
+    var privateDesktopNotifications: String
+    var privateEmailNotifications: String
+    var privateMobilePushNotifications: String
+    var privateAudioNotifications: String
+    var privateAudioNotificationsValue: String
+    var desktopNotifications: SubscriptionNotificationsStatus
+    var emailNotifications: SubscriptionNotificationsStatus
+    var mobilePushNotifications: SubscriptionNotificationsStatus
+    var audioNotifications: SubscriptionNotificationsStatus
+    var audioNotificationValue: SubscriptionNotificationsAudioValue
+    var privateOtherUserStatus: String?
+    var directMessageUser: UnmanagedUser?
+}
+
+extension UnmanagedSubscription {
+    init(_ subscription: Subscription) {
+        managedObject = subscription
```

Comment:
This seems like a nightmare to maintain. If we think this is the path forward, we should consider using https://github.com/krzysztofzablocki/Sourcery

Reply:
Yes. It is a nightmare. I'll open an issue for using Sourcery for writing all our unmanaged objects. This will perhaps allow us to fully abstract away realm from the views and view controllers.

Reply:
Thanks. Having used Sourcery before, I think it should be pretty straightforward. :+1:

Reply:
https://github.com/RocketChat/Rocket.Chat.iOS/issues/2140